File: data-structures/python/data-structures/src/list/linked_list/singly_linked_list/SinglyLinkedListNode.py
from src.list.linked_list.singly_linked_list.ISinglyLinkedListNode import ISinglyLinkedListNode


class SinglyLinkedListNode(ISinglyLinkedListNode):
    def __init__(self, data=None, next=None):
        self._data = data
        self._next = next

    @property
    def data(self):  # Overrides abstract getter
        return self._data

    @data.setter
    def data(self, val):  # Overrides abstract setter
        self._data = val

    @property
    def next(self):  # Overrides abstract getter
        return self._next

    @next.setter
    def next(self, val):  # Overrides abstract setter
        self._next = val

File: data-structures/java/data-structures/src/main/java/com/data_structures_2/list/singly_linked_list/SinglyLinkedListNode.java
package com.data_structures_2.list.singly_linked_list;

public class SinglyLinkedListNode<T> implements ISinglyLinkedListNode<T> {
    private T data;
    private ISinglyLinkedListNode<T> next;


    public SinglyLinkedListNode(T data) {
        this.data = data;
        this.next = null;
    }

    public SinglyLinkedListNode(T data, ISinglyLinkedListNode<T> next) {
        this.data = data;
        this.next = next;
    }

    public void setData(T data) {
        this.data = data;
    }

    public T getData() {
        return this.data;
    }

    public void setNext(ISinglyLinkedListNode<T> next) {
        this.next = next;
    }

    public ISinglyLinkedListNode<T> getNext() {
        return this.next;
    }
}
